plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeTransactionLogEntry.java

```diff
@@ -153,9 +167,23 @@ public CdcEntry getCDC()
         return cdcEntry;
     }
 
+    @Nullable
+    @JsonProperty
+    public SidecarEntry getSidecar()
+    {
+        return sidecar;
+    }
+
+    @Nullable
+    @JsonProperty
+    public CheckpointMetadataEntry getCheckpointMetadata()
+    {
+        return checkpointMetadata;
+    }
+
     public DeltaLakeTransactionLogEntry withCommitInfo(CommitInfoEntry commitInfo)
     {
-        return new DeltaLakeTransactionLogEntry(txn, add, remove, metaData, protocol, commitInfo, cdcEntry);
+        return new DeltaLakeTransactionLogEntry(txn, add, remove, metaData, protocol, commitInfo, cdcEntry, sidecar, checkpointMetadata);
     }
 
```

Comment:
A basic question - Don't we need the sidecard and checkpointMetadata for hashcode and equals check ? If not can we add a reason as a comment ?